app/frontend/src/features/messages/groupchats/groupChatsSlice.ts

```diff
@@ -0,0 +1,108 @@
+import { observable } from "mobx";
+import { User } from "../../../pb/api_pb";
+import { GroupChat, Message } from "../../../pb/conversations_pb";
+import { service } from "../../../service";
+
+export const initialState = observable({
+  groupChats: [] as GroupChat.AsObject[],
+  error: "",
+  loading: false,
+  groupChatView: {
+    groupChat: null as GroupChat.AsObject | null,
+    messages: [] as Message.AsObject[],
+    error: "",
+    loading: false,
+  },
+});
+
+export const fetchMessages = async (groupChat: GroupChat.AsObject) => {
+  const state = groupChatsState.groupChatView;
```

Comment:
Is this meant to be `initialState`? Where is `groupChatsState` being injected here otherwise?

Reply:
Yes, it is the same. But I did not like the name initialState. So a solution could be to rename it too groupChatsState.

Reply:
So `mobx-state-tree` is an additional constraint layer on top of `mobx` to force you to only do updates within actions only?

Reply:
No, mobx also can do that. Mobx-state-tree adds runtime type checking and snapshots.